node/sc/mainbridge_test.go

```diff
@@ -195,3 +239,113 @@ func TestMainBridge_handleMsg(t *testing.T) {
+	bc, err := testBlockChain()
+	if err != nil {
+		t.Fatal(err)
+	}
+	mBridge.blockchain = bc
+
+	// Variables will be used as return values of mockBridgePeer
+	key, _ := crypto.GenerateKey()
+	nodeID := discover.PubkeyID(&key.PublicKey)
+	peer := p2p.NewPeer(nodeID, "name", []p2p.Cap{})
+	peerID := peer.ID()
+	bridgePeerID := fmt.Sprintf("%x", peerID[:8])
+	pipe, _ := p2p.MsgPipe()
+
+	// mockBridgePeer mocks BridgePeer
+	mockCtrl := gomock.NewController(t)
+	mockBridgePeer := NewMockBridgePeer(mockCtrl)
+	mockBridgePeer.EXPECT().GetID().Return(bridgePeerID).AnyTimes()
+	mockBridgePeer.EXPECT().GetP2PPeer().Return(peer).AnyTimes()
+	mockBridgePeer.EXPECT().GetP2PPeerID().Return(peerID).AnyTimes()
+	mockBridgePeer.EXPECT().GetRW().Return(pipe).AnyTimes()
+	mockBridgePeer.EXPECT().Close().Return().AnyTimes()
+
+	// Case 1 - Error if `mBridge.peers.Len()` was equal or bigger than `mBridge.maxPeers`
+	{
+		// Set maxPeers to make the test fail
+		mBridge.maxPeers = mBridge.peers.Len()
+
+		err = mBridge.handle(mockBridgePeer)
+		assert.Equal(t, p2p.DiscTooManyPeers, err)
+	}
+	// Resolve the above failure condition by increasing maxPeers
+	mBridge.maxPeers += 5
+
+	// Case 2 - Error if handshake of BridgePeer failed
+	{
+		// Make handshake fail
+		mockBridgePeer.EXPECT().Handshake(uint64(8888), big.NewInt(8217), big.NewInt(1), bc.CurrentHeader().Hash()).Return(p2p.ErrPipeClosed).Times(1)
+
+		err = mBridge.handle(mockBridgePeer)
+		assert.Equal(t, p2p.ErrPipeClosed, err)
+	}
+	// Resolve the above failure condition by making handshake success
+	mockBridgePeer.EXPECT().Handshake(uint64(8888), big.NewInt(8217), big.NewInt(1), bc.CurrentHeader().Hash()).Return(nil).AnyTimes()
+
+	// Case 3 - Error when the same peer was registered before
+	{
+		// Pre-register a peer which will be added again
+		mBridge.peers.peers[bridgePeerID] = &baseBridgePeer{}
+
+		err = mBridge.handle(mockBridgePeer)
+		assert.Equal(t, errAlreadyRegistered, err)
+	}
+	// Resolve the above failure condition by deleting the registered peer
+	delete(mBridge.peers.peers, bridgePeerID)
+
+	// Case 4 - Error if `mBridge.handleMsg` failed
+	{
+		// Close of the peer's pipe make `mBridge.handleMsg` fail
+		_ = pipe.Close()
+
+		err = mBridge.handle(mockBridgePeer)
+		assert.Equal(t, p2p.ErrPipeClosed, err)
+	}
+}
+
+// TestMainBridge_SendRPCResponseData tests SendRPCResponseData function of MainBridge.
+// The function sends RPC response data to MainBridge's peers.
+func TestMainBridge_SendRPCResponseData(t *testing.T) {
+	// Create a MainBridge
+	mBridge := testNewMainBridge(t)
+	defer mBridge.chainDB.Close()
+
+	// Test data used as a parameter of SendResponseRPC function
+	data := []byte{0x11, 0x22, 0x33}
+
+	// mockBridgePeer mocks BridgePeer
+	mockCtrl := gomock.NewController(t)
+	mockBridgePeer := NewMockBridgePeer(mockCtrl)
+
+	// Add mockBridgePeer as a peer of `mBridge.BridgePeerSet`
+	mBridge.BridgePeerSet().peers["testID"] = mockBridgePeer
```

Comment:
How about calling `bridgePeerSet.Register`?

Reply:
Good suggestion. I changed the code to use it.